programs/magicblock/src/magic_scheduled_base_intent.rs

```diff
@@ -529,6 +570,8 @@ pub(crate) fn validate_commit_schedule_permissions(
     if parent_program_id != Some(committee_owner)
         && !signers.contains(committee_pubkey)
         && !signers.contains(&validator_id)
+        && (parent_program_id != Some(&EATA_PROGRAM_ID)
```

Comment:
That extra block is hard to understand (what does this check mean?). Maybe create an intermediate var and name it, i.e. `is_spl_token` or similar .. or comment on this line instead.